File: server/src/main/java/com/thedeveloper/qyran/models/CurrentTestsModel.java
package com.thedeveloper.qyran.models;

import com.thedeveloper.qyran.entity.CourseEntity;
import com.thedeveloper.qyran.entity.TestEntity;
import lombok.Data;

import java.util.ArrayList;
import java.util.List;

@Data
public class CurrentTestsModel {
    CourseEntity course;
    List<TestEntity> tests = new ArrayList<>();
}

File: qyran/lib/api/entity/CurrentTestsModel.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:qyran/api/entity/CourseEntity.dart';
import 'package:qyran/api/entity/TestEntity.dart';

part 'CurrentTestsModel.g.dart';

@JsonSerializable()
class CurrentTestsModel {
  CourseEntity course;
  List<TestEntity> tests;

  CurrentTestsModel({required this.course, required this.tests});

  factory CurrentTestsModel.fromJson(Map<String, dynamic> json) =>
      _$CurrentTestsModelFromJson(json);
  Map<String, dynamic> toJson() => _$CurrentTestsModelToJson(this);
}
